### PR TITLE
Issue4

### PR DESCRIPTION
**Pull request by aeftimia**
* VideoRegistry data model updated according to current data model in MongoDB
* ipfs hashes are added to video metadata
* Likes and dislikes and statistics are registered for videos and users

### CODE REVIEW

**Comment by aeftimia** [comment]
This should take care of #4

File: test/userRegistry.js [hunk 1]
```diff
@@ -72,5 +81,12 @@ contract('UserRegistry', function (accounts) {
     assert.equal(tx, false)
     tx = await userRegistry.userDislikesVideo(userAddress, videoId)
     assert.equal(tx, true)
+
+//    likes_percentage, views, likes, dislikes = await videoRegistry.getStats(videoId)
+//    assert.equal(likes_percentage, 50)
+//    assert.equal(views, 0)
+//    assert.equal(likes, 1)
+//    assert.equal(dislikes, 1)
```
Comment (jellegerbrandy) [inline test/userRegistry.js]: This assert is wrong, I think. What happend is that `userAddress` first liked the video, then disliked it. the `dislike` should  cancel the like.

File: test/userRegistry.js [hunk 2]
```diff
@@ -72,5 +81,12 @@ contract('UserRegistry', function (accounts) {
     assert.equal(tx, false)
     tx = await userRegistry.userDislikesVideo(userAddress, videoId)
     assert.equal(tx, true)
+
+//    likes_percentage, views, likes, dislikes = await videoRegistry.getStats(videoId)
+//    assert.equal(likes_percentage, 50)
+//    assert.equal(views, 0)
+//    assert.equal(likes, 1)
```
Comment (jellegerbrandy) [inline test/userRegistry.js]: uncomment this test

File: contracts/paratii/UserRegistry.sol [hunk 3]
```diff
@@ -71,9 +80,11 @@ contract UserRegistry is Ownable {
       if (_liked) {
         video.liked = true;
         video.disliked = false;
+        videoRegistry.likeVideo(_videoId);
       } else {
         video.liked = false;
         video.disliked = true;
+        videoRegistry.dislikeVideo(_videoId);
```
Comment (jellegerbrandy) [inline contracts/paratii/UserRegistry.sol]: we need some more logic here. If the user first liked a video, then diskliked it, the like counter in the videoregistry should be diminished.

Comment (jellegerbrandy) [reply contracts/paratii/UserRegistry.sol]: and vice versa mutatis mutandis.

File: test/userRegistry.js [hunk 4]
```diff
@@ -8,50 +8,53 @@ contract('UserRegistry', function (accounts) {
...
+    await setupParatiiContracts()
     await expectError(async function() {
-      await userRegistry.registerUser(web3.eth.accounts[1], name, email, {from: web3.eth.accounts[2]})
+      await userRegistry.registerUser(web3.eth.accounts[1], name, email, avatar, {from: web3.eth.accounts[2]})
     })
 
     return
     // now register a user and try to unregister it from another account
-    await userRegistry.registerUser(web3.eth.accounts[1], name, email)
+    await userRegistry.registerUser(web3.eth.accounts[1], name, email, avatar)
```
Comment (jellegerbrandy) [inline test/userRegistry.js]: this test, after the return statement, is not running

**Comment by aeftimia** [comment]
OK @jellegerbrandy, this latest commit should address all of your comments. I have an exhaustive set of tests for pairwise all combinations of likes and dislikes for up to two users. This tests cumulative likes/dislike (different users), canceling likes/dislikes (same user votes differently), and single likes/dislikes with registration.

**Comment by aeftimia** [comment]
#4 branch is now rebased against `dev`; conflicts resolved.

**Comment by jellegerbrandy** [comment]
Great work, alex, sorry for the delay

**Pull request merged**